src/KafkaFlow.Abstractions/IConsumerContext.cs

```diff
@@ -49,14 +55,43 @@ public interface IConsumerContext
         DateTime MessageTimestamp { get; }
 
         /// <summary>
-        /// Gets or sets a value indicating whether if the framework should store the current offset in the end when auto store offset is used
+        /// Gets or sets a value indicating whether if the framework should auto complete the message in the end
+        /// </summary>
+        bool AutoMessageCompletion { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether if the message offset must be stored when the message is marked as completed
+        /// </summary>
+        bool StoreOffsetOnCompletion { get; set; }
+
+        /// <summary>
+        /// Gets an instance of IDependencyResolver which provides methods to resolve dependencies.
+        /// This instance is tied to the consumer scope, meaning it is capable of resolving dependencies
+        /// that are scoped to the lifecycle of a single consumer.
+        /// </summary>
+        IDependencyResolver ConsumerDependencyResolver { get; }
+
+        /// <summary>
+        /// Gets an instance of IDependencyResolver which provides methods to resolve dependencies.
+        /// This instance is tied to the worker scope, meaning it is capable of resolving dependencies
+        /// that are scoped to the lifecycle of a single worker.
+        /// </summary>
+        IDependencyResolver WorkerDependencyResolver { get; }
```

Comment:
Because each container instance has a different lifetime, their scopes are disposed in different moments. For the Microsoft DI consumer's instances or worker's instances both are "scoped", KafkaFlow manually manages these scopes to create custom lifetimes, such as consumer lifetime and worker lifetime.